frontend/components/cards/Category.tsx

```diff
@@ -0,0 +1,92 @@
+// components/CategoryChips.tsx
+import { Colors, Theme } from "@/constants/theme";
+import React, { useState } from "react";
+import {
+  FlatList,
+  TouchableOpacity,
+  Text,
+  StyleSheet,
+  View,
+  Platform,
+} from "react-native";
+
+type Props = {
+  data: string[];
+  onChange?: (value: string | null) => void; // returns selected or null if deselected
+  initial?: string | null;
+};
+
+export default function CategoryChips({
+  data,
+  onChange,
+  initial = null,
+}: Props) {
+  const [active, setActive] = useState<string | null>(initial);
+
+  const toggle = (item: string) => {
+    const next = active === item ? null : item;
+    setActive(next);
+    onChange?.(next);
+  };
+
+  return (
+    <FlatList
+      data={data}
+      keyExtractor={(i) => i}
+      horizontal
+      showsHorizontalScrollIndicator={false}
+      contentContainerStyle={styles.row}
+      ListHeaderComponent={<View style={{ width: 12 }} />}
+      ListFooterComponent={<View style={{ width: 12 }} />}
+      renderItem={({ item }) => {
```

Comment:
Separate mo to as function 
```suggestion
      function CategoryChipsItemView({ item }) {
        ...rest of the code
      }
      renderItem={CategoryChipsItemView}
```

Reply:
Function separated.